### ISSUE TITLE
Deprecations warnings from Markdown

### ISSUE THREAD

**Issue by benjaoming** [django-wiki/django-wiki#1099]
We have a series of deprecation warnings that should be fixed.

They are easy to reproduce by running this test:

```
# Install test requirements
pip install -e '.[test]'
# Run this test
pytest tests/plugins/editsection/test_editsection.py
```

The deprecation warnings:

```
  /home/user/.virtualenvs/django-wiki/lib/python3.8/site-packages/markdown/core.py:124: DeprecationWarning: The 'md_globals' parameter of 'wiki.plugins.editsection.markdown_extensions.EditSectionExtension.extendMarkdown' is deprecated.
    ext._extendMarkdown(self)

tests/plugins/editsection/test_editsection.py::EditSectionTests::test_broken_content
  /home/user/.virtualenvs/django-wiki/lib/python3.8/site-packages/markdown/inlinepatterns.py:206: DeprecationWarning: Flags not at the start of the expression '^(.*?)(?:(?im)\\[imag' (truncated)
    self.compiled_re = re.compile(r"^(.*?)%s(.*)$" % pattern,

tests/plugins/editsection/test_editsection.py::EditSectionTests::test_broken_content
  /home/user/.virtualenvs/django-wiki/lib/python3.8/site-packages/markdown/inlinepatterns.py:206: DeprecationWarning: Flags not at the start of the expression '^(.*?)(?:(?im)\\[imag' (truncated)
    self.compiled_re = re.compile(r"^(.*?)%s(.*)$" % pattern,

  /media/archive/code/django-wiki/src/wiki/plugins/images/markdown_extensions.py:33: DeprecationWarning: Using the add method to register a processor or pattern is deprecated. Use the `register` method instead.
    md.inlinePatterns.add("dw-images", ImagePattern(IMAGE_RE, md), ">link")
```

**Comment by iqbalabd** [comment]
I have made a pull request [here](https://github.com/django-wiki/django-wiki/pull/1125) for this issue.

**Comment by benjaoming** [comment]
Fixed in #1125 - thanks :pray:
